db.sql.editor/test/unit/src/org/netbeans/modules/db/sql/editor/completion/SelectCompletionQueryTest.java

```diff
@@ -125,11 +125,11 @@ public void runTest() throws Exception {
 
     public void testCompletion() throws Exception {
         StringBuilder sqlData = new StringBuilder();
-        List<String> modelData = new ArrayList<String>();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(SelectCompletionQueryTest.class.getResource(getName() + ".test").openStream(), "utf-8"));
-        try {
+        List<String> modelData = new ArrayList<>();
+        try (InputStream is = SelectCompletionQueryTest.class.getResourceAsStream(getName() + ".test");
+                BufferedReader reader = new BufferedReader(new InputStreamReader(is, "utf-8"))) {
             boolean separatorRead = false;
-            for (String line; (line = reader.readLine()) != null;) {
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
```

Comment:
Wasn't the previous `for` line better?

Reply:
I consider the previous one less readable. The previous version mixed initialization, loop advance and checking and these parts are now separated into their corresponding sections.

Reply:
I believe the canonical trick is to have

```java
String line;
while((line = reader.readLine()) != null) {
  ...
}
```

And the original `for` just mixed those to limit the scope of the `line` variable to the `for` block... which is nice.

Your `for` repeats the `reader.readLine()` call which seems confusing.

Reply:
I'll revert my change to this loop and then push to master.